scipy/interpolate/_rbfinterp.py

```diff
@@ -387,6 +387,64 @@ def __init__(self, y, d,
         self.epsilon = epsilon
         self.powers = powers
 
+    def _chunk_evaluator(
+            self,
+            x,
+            y,
+            shift,
+            scale,
+            coeffs,
+            memory_budget=None
+    ):
+        """
+        Evaluate the interpolation why controlling memory consumption.
+        We chunk the input if we need more memory than specified.
+
+        Parameters
+        ----------
+        x : (Q, N) float ndarray
+            array of points on which to evaluate
+        y: (P, N) float ndarray
+            array of points on which we know function values
+        shift: (N, ) array
+            Domain shift used to create the polynomial matrix.
+        scale : (N,) float ndarray
+            Domain scaling used to create the polynomial matrix.
+        coeffs: (Q, P+R) float ndarray
+            Coefficients in front of basis functions
+        memory_budget: int
+            Total amount of memory (in units of sizeof(float)) we wish
+            to devote for storing the array of coefficients for
+            interpolated points. If we need more memory than that, we
+            chunk the input.
+
+        Returns
+        -------
+        (Q, ) float ndarray
+        Interpolated array
+        """
+        nx, ndim = x.shape
+        if self.neighbors is None:
+            nnei = len(y)
+        else:
+            nnei = self.neighbors
+        # in each chunk we consume the same space we already occupy
+        chunksize = memory_budget // ((self.powers.shape[0] + nnei)) + 1
+        if chunksize <= nx:
+            out = np.empty((nx, self.d.shape[1]), dtype=float)
+            for i in range(0, nx, chunksize):
+                # I had to use copy() here because pythran doesnt seem to
+                # accept views
+                vec = _evaluate(x[i:i + chunksize, :].copy(), y, self.kernel,
```

Comment:
the copies are not needed. Pythran just throws an error when the array is not contiguous. `x` is c contiguous, so you can pass its slices to `_evaluate` without issue.  `coeffs` is fortran-contiguous, so pythran does throw an error when you slice it. However, `_evaluate` no longer users `coeffs`, so there is no longer any need to pass it to `_evaluate`.

Reply:
fixed in new version